### PR TITLE
[GPU] Add disable winograd convolution in execute_config.

### PR DESCRIPTION
**Pull request by hyunback**
### Tickets:
 - *115819*

### CODE REVIEW

**Comment by hyunback** [comment]
On going reviewing winograd convolution itself

File: src/inference/include/ie/gpu/gpu_config.hpp [hunk 1]
```diff
@@ -130,6 +130,14 @@ INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_GPU_CONFIG_KEY(MAX_NUM_THREADS);
  * Thus, this key should be turned off if graph loading time is considered to be most important target to optimize.*/
 INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_GPU_CONFIG_KEY(ENABLE_LOOP_UNROLLING);
 
+/**
+ * @brief Turning on this key disable winograd convoltuion in iGPU.
```
Comment (isanghao) [inline src/inference/include/ie/gpu/gpu_config.hpp:134]: * `disable` -> `disables`
* `convoltuion` -> `convolution`

File: src/inference/include/ie/gpu/gpu_config.hpp [hunk 2]
```diff
@@ -130,6 +130,14 @@ INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_GPU_CONFIG_KEY(MAX_NUM_THREADS);
...
+ * Winograd's algorithm for convolutions can greatly reduce the number of arithmetic operations.
+ * However a large reduction in floating point (FP) operations in these algorithmscan result in reduced FP accuracy of
+ * the result. Default value is false.
```
Comment (isanghao) [inline src/inference/include/ie/gpu/gpu_config.hpp:137]: I think these three lines are too much detail. We can simplify mention like this: `Winograd convolution has different characteristics for accuracy and performance compared to other convolution implementations.`

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp [hunk 3]
```diff
@@ -577,6 +577,8 @@ bool layout_optimizer::convolution_byxf_opt(const layout& input_layout,
             return false;
     }
 
+    bool disable_winograd_conv = node.get_program().get_config().get_property(ov::intel_gpu::disable_winograd_convolution);
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/graph/layout_optimizer.cpp:580]: nit: what about checking this condition from `should_use_winograd_2x3_s1` function? I think that will look more straightforward.

File: src/bindings/python/tests/test_runtime/test_properties.py [hunk 4]
```diff
@@ -328,6 +328,11 @@ def test_properties_ro(ov_property_ro, expected_value):
             "GPU_ENABLE_LOOP_UNROLLING",
             ((True, True),),
         ),
+        (
+            properties.intel_gpu.disable_winograd_convolution,
+            "GPU_DISABLE_WINOGRAD_CONVOLUTION",
+            ((True, True),),
+        ),
```
Comment (isanghao) [inline src/bindings/python/tests/test_runtime/test_properties.py:335]: please update documentation as well

**Review by isanghao — CHANGES_REQUESTED** [review]

File: src/inference/include/ie/gpu/gpu_config.hpp (hunk 1 above)
Comment (hyunback) [reply src/inference/include/ie/gpu/gpu_config.hpp]: Applied

File: src/inference/include/ie/gpu/gpu_config.hpp (hunk 2 above)
Comment (hyunback) [reply src/inference/include/ie/gpu/gpu_config.hpp]: Applied

File: src/plugins/intel_gpu/src/graph/layout_optimizer.cpp (hunk 3 above)
Comment (hyunback) [reply src/plugins/intel_gpu/src/graph/layout_optimizer.cpp]: Applied

File: src/bindings/python/tests/test_runtime/test_properties.py (hunk 4 above)
Comment (hyunback) [reply src/bindings/python/tests/test_runtime/test_properties.py:335]: Applied

**Review by isanghao — APPROVED** [review]

File: src/inference/include/ie/gpu/gpu_config.hpp [hunk 5]
```diff
@@ -130,6 +130,13 @@ INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_GPU_CONFIG_KEY(MAX_NUM_THREADS);
  * Thus, this key should be turned off if graph loading time is considered to be most important target to optimize.*/
 INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_GPU_CONFIG_KEY(ENABLE_LOOP_UNROLLING);
 
+/**
+ * @brief Turning on this key disables winograd convolution in iGPU.
+ * Winograd convolution has different characteristics for accuracy and performance compared to other convolution
+ * implementations. Default value is false.
+ */
+INFERENCE_ENGINE_1_0_DEPRECATED DECLARE_GPU_CONFIG_KEY(DISABLE_WINOGRAD_CONVOLUTION);
```
Comment (vladimir-paramuzov) [inline src/inference/include/ie/gpu/gpu_config.hpp:138]: Don't add anything to the old api please

File: src/inference/include/openvino/runtime/intel_gpu/properties.hpp [hunk 6]
```diff
@@ -63,6 +63,14 @@ static constexpr Property<std::map<std::string, uint64_t>, PropertyMutability::R
  */
 static constexpr Property<bool> enable_loop_unrolling{"GPU_ENABLE_LOOP_UNROLLING"};
 
+/**
+ * @brief Turning on this key disables winograd convolution in iGPU.
```
Comment (vladimir-paramuzov) [inline src/inference/include/openvino/runtime/intel_gpu/properties.hpp:67]: No need to mention iGPU here - it's implementation detail.

File: src/inference/include/ie/gpu/gpu_config.hpp (hunk 5 above)
Comment (hyunback) [reply src/inference/include/ie/gpu/gpu_config.hpp]: removed it

File: src/inference/include/openvino/runtime/intel_gpu/properties.hpp (hunk 6 above)
Comment (hyunback) [reply src/inference/include/openvino/runtime/intel_gpu/properties.hpp]: Applied

**Pull request merged**